tcms/static/js/tcms_actions.js

```diff
@@ -1392,20 +1391,25 @@ function exportCase(url, form, table) {
 
 var printableCases = exportCase;
 
-function validateIssueID(bugSystemId, bugId) {
-  if (bugSystemId == 1) {
-    var result = /^\d{1,7}$/.test(bugId);
-    if (!result) {
-      window.alert(default_messages.alert.invalid_bug_id);
-    }
-    return result;
-  } else if (bugSystemId == 2) {
-    var result = /^[A-Z0-9]+-\d+$/.test(bugId);
-    if (!result) {
-      window.alert(default_messages.alert.invalid_jira_id);
-    }
-    return result;
+function validateIssueID(bugRegExp, bugId) {
+  // if bugRegExp is empty string then all input is valid!
+  if (!bugRegExp) {
```

Comment:
The bug ID regular expression must be defined in the database and it also must be available in user's web browser in order to check the bug ID before sending to server. So, when run at this line, ``bugRegExp`` should not be empty and if it is, Nitrate should refuse to add this bug ID. Checking whether ``bugRegExp`` is empty string explicitly is a better way than only using ``!``.

```javascript
if (bugRegExp === '') {
     return false;
}
```